image/templates/helm/shared/templates/02-scanner-v4-07-indexer-deployment.yaml

```diff
@@ -82,17 +82,18 @@ spec:
             drop: ["NET_RAW"]
           runAsNonRoot: true
           runAsUser: 65534
-# TODO(mclamei): Enable when the grpc healthcheck in indexer works.
-#
-# {{ if semverCompare ">= 1.24.0" ._rox._apiServer.version }}
-#         readinessProbe:
-#           grpc:
-#             port: 8443
-#           timeoutSeconds: 10
-#           periodSeconds: 10
-#           failureThreshold: 6
-#           successThreshold: 1
-# {{- end }}
+        livenessProbe:
+          httpGet:
+            scheme: HTTPS
+            path: /health/liveness
+            port: 9443
```

Comment:
> What is the motivation for adding a livenessProbe here (and in matcher yaml)? 

Liveness will swing the POD if it's unresponsive. I agree our liveness is shallow, but still.

> If the pod is slow to start for whatever reason (perhaps on under provisioned nodes) - this may keep killing the pod.

The initialization process is expected to be completed in a few seconds. However, there might be an exception in the case of migrations, which ClairCore controls. In this patch, migrations happen before the liveness endpoint is available. We could consider moving the liveness endpoint earlier in the initialization stack.

> > If the process in your container is able to crash on its own

Is that always true for Matcher and Indexer? If yes, I would be OK removing liveness.

Reply:
@dcaravel I removed the liveness check due to the current DB connection retry strategy. I'm resolving this due to that, but feel free to re-open if you want to follow-up on something.